src/Microsoft.AspNetCore.Http.Connections.Client/Internal/AccessTokenHttpMessageHandler.cs

```diff
@@ -11,16 +10,16 @@ namespace Microsoft.AspNetCore.Http.Connections.Client.Internal
 {
     internal class AccessTokenHttpMessageHandler : DelegatingHandler
     {
-        private readonly HttpConnection _httpConnection;
+        private readonly HttpConnectionOptions _httpConnectionOptions;
 
-        public AccessTokenHttpMessageHandler(HttpMessageHandler inner, HttpConnection httpConnection) : base(inner)
+        public AccessTokenHttpMessageHandler(HttpMessageHandler inner, HttpConnectionOptions httpConnectionOptions) : base(inner)
         {
-            _httpConnection = httpConnection;
+            _httpConnectionOptions = httpConnectionOptions;
         }
 
         protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
         {
-            var accessToken = await _httpConnection.GetAccessTokenAsync();
+            var accessToken = (_httpConnectionOptions.AccessTokenProvider != null) ? await _httpConnectionOptions.AccessTokenProvider() : null;
```

Comment:
Either `_httpConnectionOptions.AccessTokenProvider?.Invoke` or assign to local variable. Avoid race issues with another thread setting AccessTokenProvider to null.